libraries/ESP8266WiFi/src/ESP8266WiFiSTA.h

```diff
@@ -49,6 +49,7 @@ class ESP8266WiFiSTAClass: public LwipIntf {
 
         // two and one parameter version. 2nd parameter is DNS like in Arduino
         // IPv4 only
+        [[deprecated("It is discouraged to use this 1 or 2 parameters network configuration legacy function config(ip[,dns]) because it sets the router (and ommited dns) to an arbitrary address")]]

```

Comment:
It doesn't set the router, but either sets the gateway, or makes assumptions about the router's IP-address.